src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs

```diff
@@ -19,6 +27,102 @@ public class RoslynCodeTaskFactory_Tests
     {
         private const string TaskName = "MyInlineTask";
 
+        [Fact]
+        public void InlineTaskWithAssemblyPlatformAgnostic()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                TransientTestFolder folder = env.CreateFolder(createFolder: true);
+                string location = Assembly.GetExecutingAssembly().Location;
+                TransientTestFile inlineTask = env.CreateFile(folder, "5106.proj", @$"
+<Project>
+
+  <UsingTask TaskName=""MyInlineTask"" TaskFactory=""RoslynCodeTaskFactory"" AssemblyFile=""$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll"">
+    <Task>
+      <Reference Include=""{ Path.Combine(Path.GetDirectoryName(location), "..", "..", "..", "Samples", "Dependency", location.Contains("Debug") ? "Debug" : "Release", "net472", "Dependency.dll") }"" />
```

Comment:
I think I'd do `#if DEBUG` instead of the path-contains check (it's more reliable in the case that e.g. some dev has all their source in `Q:\Debug\msbuild`) but that's pretty nitpicky.

Reply:
This is doable but a bit fiddly because I don't think you can put a #if in the middle of a string. I can make it three strings and concat them, I guess.